source/basic.d.ts

```diff
@@ -65,3 +65,45 @@ export interface ObservableLike {
 	subscribe(observer: (value: unknown) => void): void;
 	[Symbol.observable](): ObservableLike;
 }
+
+/**
+Creates an opaque type, which hides its internal details from the public, and can only be created by being used explicitly.
+
+Read more about opaque types here, this link also includes an example that shows how they can be useful:
+  [Codemix](https://codemix.com/opaque-types-in-javascript/)
+
+There have been several discussions about adding this feature to TypeScript via the `opaque type` operator, similar to how Flow does it. Unfortunately, nothing has (yet) moved forward:
+  1. [](https://github.com/Microsoft/TypeScript/issues/15408)
+  2. [](https://github.com/Microsoft/TypeScript/issues/15807)
+
+Please note that the type (the `Type` parameter) can be anything. It doesn't have to be an object.
+
+@example
+```ts
+import {Opaque} from 'type-fest';
+
+type AccountNumber = Opaque<number>;
+type AccountBalance = Opaque<number>;
+
+function createAccountNumber (): AccountNumber {
+  return 2 as AccountNumber;
```

Comment:
Use tabs instead of spaces:
```suggestion
	return 2 as AccountNumber;
```